pkg/apis/maistra/v2/gateways.go

```diff
@@ -112,6 +112,9 @@ type ClusterIngressGatewayConfig struct {
 	// MeshExpansionPorts define the port set used with multi-cluster/mesh expansion
 	// +optional
 	MeshExpansionPorts []corev1.ServicePort `json:"meshExpansionPorts,omitempty"`
+	// RouteEnabled specifies whether to create a Route (route.openshift.io/v1) for the istio-ingressgateway component
+	// +optional
+	RouteEnabled *bool `json:"route,omitempty"`
```

Comment:
I think a more complete solution might be to add an `Ingress` field of type `ComponentIngressConfig`.

Reply:
Agreed, and `ingress.ingress` is odd.  Perhaps then,
```go
    RouteConfig *ClusterIngressGatewayConfig `json:"routeConfig,omitempty"`
```

Reply:
Don't you mean ```RouteConfig *ComponentIngressConfig `json:"routeConfig,omitempty"` ```?

Reply:
I assumed that you suggest to use type `ComponentIngressConfig`, but to handle only field `enabled` for now. Am I right?

Reply:
Yes, and that type includes an `enabled` field.

Reply:
For reference, https://github.com/maistra/istio-operator/blob/maistra-2.2/pkg/apis/maistra/v2/runtime.go#L312

Reply:
@rcernich I don't understand, why do we want to add this whole struct? The issue was about adding a bool enable/disable field, now we're adding a bunch of fields with no functionality?

Reply:
This approach enables us to make other route properties customizable in the future, e.g. configure TLS.

Reply:
sure, I get why we would use a struct with `Enablement`, but why reuse an existing one and not support the fields? Can't we just have an empty struct, with just a bool and add fields if/when we actually implement them?